src/client/client.ts

```diff
@@ -321,35 +364,47 @@ class KafkaJsClient implements Client {
     }
 
     async deleteTopic(deleteTopicRequest: DeleteTopicRequest): Promise<void> {
+        if (!this.kafkaAdminClient) {
+            throw new Error('TODO');
+        }
         return await this.kafkaAdminClient.deleteTopics({
             topics: deleteTopicRequest.topics,
             timeout: deleteTopicRequest.timeout
         });
     }
 
     dispose() {
+        if (this.kafkaAdminClient) {
         this.kafkaAdminClient.disconnect();
+        }
     }
 }
 
 export const createClient = (cluster: Cluster, workspaceSettings: WorkspaceSettings): Client => new EnsureConnectedDecorator(
     new KafkaJsClient(cluster, workspaceSettings));
 
-export const createKafka = (connectionOptions: ConnectionOptions): Kafka => {
-    let kafkaJsClient: Kafka;
+export const createKafka = async (connectionOptions: ConnectionOptions): Promise<Kafka> => {
+    const provider = getClusterProvider(connectionOptions.clusterProviderId);
+    if (!provider) {
+        //throw new Error(`Cannot find cluster provider for '${connectionOptions.clusterProviderId}' ID.`);
+        return createDefaultKafka(connectionOptions);
```

Comment:
that's a bad idea, it creates a broken client for this particular cluster

Reply:
fixed, now it throw an error.